### PR TITLE
Qt - Boot (S)ELF file - Path Icons

### PR DESCRIPTION
**Pull request by scribam**
Check if the pth is a valid directory before to look for icons.
Otherwise, the function `fs::is_file(ico)` will crash while checking for a path like this one:
`/my/self/file.self/ICON0.PNG`
where `/my/self/file.self` is a path to a self file to boot.

Should fix the error reported by @mrcmunir inside the issue 2821.

### CODE REVIEW

**Comment by Megamouse** [comment]
sounds reasonable. I didn't think this could crash

**Comment by mrcmunir** [comment]
@scribam Thanks fixed now my problem  sprx manually loading fine.

File: rpcs3/rpcs3qt/main_window.cpp [hunk 1]
```diff
@@ -131,35 +131,38 @@ void main_window::SetAppIconFromPath(const std::string path)
 	// get Icon for the gs_frame from path. this handles presumably all possible use cases
 	QString qpath = qstr(path);
 	std::string icon_list[] = { "/ICON0.PNG", "/PS3_GAME/ICON0.PNG" };
-	std::string path_list[] = { path, sstr(qpath.section("/", 0, -2)) ,sstr(qpath.section("/", 0, -3)) };
+	std::string path_list[] = { path, sstr(qpath.section("/", 0, -2)), sstr(qpath.section("/", 0, -3)) };
 	for (std::string pth : path_list)
 	{
-		for (std::string ico : icon_list)
+		if (fs::is_dir(pth))
```
Comment (Nekotekina) [inline rpcs3/rpcs3qt/main_window.cpp]: Can you use if !is_dir then continue?

Comment (scribam) [inline rpcs3/rpcs3qt/main_window.cpp]: Done.

File: rpcs3/rpcs3qt/main_window.cpp [hunk 2]
```diff
@@ -131,9 +131,11 @@ void main_window::SetAppIconFromPath(const std::string path)
 	// get Icon for the gs_frame from path. this handles presumably all possible use cases
 	QString qpath = qstr(path);
 	std::string icon_list[] = { "/ICON0.PNG", "/PS3_GAME/ICON0.PNG" };
-	std::string path_list[] = { path, sstr(qpath.section("/", 0, -2)) ,sstr(qpath.section("/", 0, -3)) };
+	std::string path_list[] = { path, sstr(qpath.section("/", 0, -2)), sstr(qpath.section("/", 0, -3)) };
 	for (std::string pth : path_list)
 	{
+        if (!fs::is_dir(pth)) continue;
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp]: spaces

Comment (scribam) [inline rpcs3/rpcs3qt/main_window.cpp]: Done.

**Comment by hcorion** [comment]
Would appreciate this getting merged!

**Pull request merged**